osu.Game/Screens/Play/HUD/HoldForMenuButton.cs

```diff
@@ -60,14 +69,15 @@ public HoldForMenuButton()
             AutoSizeAxes = Axes.Both;
         }
 
-        [Resolved]
-        private OsuConfigManager config { get; set; }
-
-        private Bindable<double> activationDelay;
-
         protected override void LoadComplete()
         {
-            activationDelay = config.GetBindable<double>(OsuSetting.UIHoldActivationDelay);
+            if (player?.Configuration.AllowRestart == false)
+            {
+                activationDelay.Value = PopupDialogDangerousButton.DANGEROUS_HOLD_ACTIVATION_DELAY;
+            }
+            else
+                config.BindWith(OsuSetting.UIHoldActivationDelay, activationDelay);
```

Comment:
I was really confused as to why this was reimplementing the hold-to-confirm logic locally and only later realised that this is keeping a local copy for text display purposes... This is an argument in favour of removing the logic from the abstract class and permitting consumers to configure on their own.

Reply:
Yep, exactly.